spec/Section 5 -- Validation.md

```diff
@@ -780,6 +801,27 @@ fragment missingRequiredArg on Arguments {
 }
 ```
 
+#### Oneof Fields Have Exactly One Argument
+
+* For each {operation} in {document}:
+  * Let {oneofFields} be all Oneof Fields transitively included in the {operation}.
+  * For each {oneofField} in {oneofFields}:
+    * Let {arguments} be the arguments provided by {oneofField}.
+    * {arguments} must contain exactly one entry.
+    * Let {argument} be the sole entry in {arguments}.
+    * Let {value} be the value of {argument}.
+    * {value} must not be the {null} literal.
+    * If {value} is a variable:
+      * Let {variableName} be the name of {variable}.
+      * Let {variableDefinition} be the {VariableDefinition} named {variableName} defined within {operation}.
+      * Let {variableType} be the expected type of {variableDefinition}.
+      * {variableType} must be a non-null type.
```

Comment:
🤔  what about nullable variables with non-null default values? Do we permit those? Or you just can't use nullable variables in `oneOf`s, full stop?

Reply:
A nullable variable with a non-null default value can be explicitly be set to the `null` value (the default is only used when no value is provided for the variable), and thus we must not permit nullable variables here. Here's the relevant spec text:

https://spec.graphql.org/draft/#sel-IANLLFCFFDHAAHRDBFBGB_ub

Trivia: A non-nullable variable may still have a default value, in this case the variable is _optional_ within the request, but is guaranted to be _non-null_ during execution.